.github/actions/setup/action.yml

```diff
@@ -0,0 +1,48 @@
+name: Setup
```

Comment:
This is a reusable setup action we should use in other workflows

Reply:
To keep pr focused I didn't change other workflows to using it yet but will in future pr